lib/cisco_node_utils/pim_group_list.rb

```diff
@@ -0,0 +1,132 @@
+# PIM feature
+# Provides configuration of PIM configuration of rp-addresses with group-lists
+#
+# Smitha Gopalan, November 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#-------------------------------------------------------------------------------
+# CLI: ip pim rp-address <rp-address> group-list <group> (under different VRFs)
+#-------------------------------------------------------------------------------
+
+require_relative 'node_util'
+require 'pp'
+
+module Cisco
+  # node_utils class for pim grouplist
+  class PimGroupList < NodeUtil
+    attr_reader :rp_addr, :vrf, :group
+
+    # Constructor with grouplist and vrf
+    # ----------------------------------
+    def initialize(rp_addr, group, vrf='default', instantiate=true)
+      fail ArgumentError unless vrf.is_a? String
+      fail ArgumentError unless vrf.length > 0
+      @rp_addr = rp_addr
+      @group = group
+      @vrf = vrf
+      if @vrf == 'default'
+        @get_args = @set_args = { state: '', addr: @rp_addr, group: @group }
+      else
+        @get_args = @set_args = { state: '', addr: @rp_addr,
+                                  vrf: @vrf, group: @group }
+      end
+      create if instantiate
+    end
+
+    # Create a hash of [rp-addr,grouplist]=>vrf mappings
+    # --------------------------------------------------
```

Comment:
You'll need to update this method for v4/v6. The ace code already does this so your logic can be similar:

https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/ace.rb#L28